### ISSUE TITLE
Different environment instances share globals

### ISSUE THREAD

**Issue by thepechinator** [mozilla/nunjucks#573]
Wanted to see if this was intended behavior. When I call `nunjucks.configure` multiple times and get different Environment instances: if I add a global to the first instance, it will also be tied to any other instance that is created afterwards.

Example:
```javascript
var env = nunjucks.configure();
env.addGlobal('test', 'value');

// Looking at the code, this should return a new instance of Environment
var env2 = nunjucks.configure();
// will return 'value' when I was expecting it to say 'global not found'
env2.getGlobal('test');
```

Most people will probably only need one Environment in most cases, but this is something I didn't realize until I ran into it. Also, kind of related and out of curiosity, is there a good reason there's no `removeGlobal` or `removeFilter` method? Thanks!

**Comment by rhengles** [comment]
Is the last line a typo? You're reading from the same `env` where you called `addGlobal`.

**Comment by thepechinator** [comment]
Oh yeah, good catch. Thanks, I edited it.

**Comment by carljm** [comment]
I don't have time to look into this and verify it at the moment, but if in fact this is the behavior, IMO it is clearly a bug and should be fixed. Environments should be independent.

Also I have no problem with adding `removeFilter` and `removeGlobal`, just nobody has bothered to do so yet. That should be opened as a separate issue (or just a PR).

**Comment by devoidfury** [comment]
Just confirming, globals are global to nunjucks -- it's essentially just the exports object from src/globals.js

**Comment by thepechinator** [comment]
Thanks for the quick response. I went ahead and made a PR for the globals.

**Comment by carljm** [comment]
Fixed with merge of #574.
